### PR TITLE
Release v4.2.1

### PR DESCRIPTION
**Pull request by vcerenu**
Release Issue: #512 
This PR adds the latest changes to branch 4.2.1

### CODE REVIEW

File: production_cluster/kibana_ssl/generate-self-signed-cert.sh [hunk 1]
```diff
@@ -9,4 +9,5 @@ then
     exit
 else
     openssl req -x509 -batch -nodes -days 365 -newkey rsa:2048 -keyout key.pem -out cert.pem
+    chown -R 1000:1000 .
```
Comment (alberpilot) [inline production_cluster/kibana_ssl/generate-self-signed-cert.sh:12]: Please be specific with those files who need the chown.

**Review by alberpilot — CHANGES_REQUESTED** [review]
GJ, minor changes. Please review.

**Review by alberpilot — APPROVED** [review]
LGTM

File: production_cluster/kibana_ssl/generate-self-signed-cert.sh (hunk 1 above)
Comment (vcerenu) [reply production_cluster/kibana_ssl/generate-self-signed-cert.sh]: The function is modified to assign the permissions to all the .pem files in the directory

**Pull request merged**